challenges/back-end-challenge/flask_app.py

```diff
@@ -0,0 +1,52 @@
+
+# A very simple Flask Hello World app for you to get started with...
+
+from flask import Flask, request, Response
+import json
+from os import path
+
+app = Flask(__name__)
+
+sorted_data = []
+
+@app.route('/data', methods=["GET","POST","PATCH"])
+def hello_world():
+
+    try:
+        if request.method == "POST":
+
+            data = request.json
+            if data is None or len(data) != 500:
+                return Response("Exactly 500 numbers are required, please try again.", status=400)
+            data.sort()
```

Comment:
If we discover that our client is frequently POSTing new numbers but rarely using the GET to retrieve the numbers, how could we optimize our API?

Reply:
We can store the unsorted numbers in the POST request and sort them in GET request. That way we will improve our performance of the API as sorting is the bottleneck to our problem.

Basically, `data.sort()` function will be in GET request instead of POST.